openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/LineageRepository.java

```diff
@@ -81,13 +93,55 @@ public void addLineage(AddLineage addLineage) {
 
     // Finally, add lineage relationship
     dao.relationshipDAO()
-        .insert(
-            from.getId(),
-            to.getId(),
-            from.getType(),
-            to.getType(),
-            Relationship.UPSTREAM.ordinal(),
-            detailsJson);
+        .insert(from.getId(), to.getId(), from.getType(), to.getType(), Relationship.UPSTREAM.ordinal(), detailsJson);
+    addLineageToSearch(from, to, addLineage.getEdge().getLineageDetails());
+  }
+
+  private void addLineageToSearch(EntityReference fromEntity, EntityReference toEntity, LineageDetails lineageDetails) {
+    IndexMapping sourceIndexMapping = Entity.getSearchRepository().getIndexMapping(fromEntity.getType());
+    String sourceIndexName = sourceIndexMapping.getIndexName();
```

Comment:
@07Himank I've merged clusterAlias to allow multiple OM instances to run against single search cluster. Take a look at those changes and update this code